tiledb/sm/c_api/tiledb.cc

```diff
@@ -1742,6 +1771,107 @@ int tiledb_query_get_status(
 /*              ARRAY             */
 /* ****************************** */
 
+int tiledb_array_alloc(
+    tiledb_ctx_t* ctx, const char* array_uri, tiledb_array_t** array) {
+  if (sanity_check(ctx) == TILEDB_ERR)
+    return TILEDB_ERR;
+
+  // Create array struct
+  *array = new (std::nothrow) tiledb_array_t;
+  if (*array == nullptr) {
+    auto st =
+        tiledb::sm::Status::Error("Failed to allocate TileDB array object");
+    LOG_STATUS(st);
+    save_error(ctx, st);
+    return TILEDB_OOM;
+  }
+
+  // Set array URI
+  (*array)->array_uri_ = tiledb::sm::URI(array_uri);
```

Comment:
This will be caught later when the array gets opened, but we could add a check. I am planning to address invalid inputs for the entire C API in a new PR.